bundles/org.openhab.binding.gridbox/src/main/java/org/openhab/binding/gridbox/internal/api/GridBoxApi.java

```diff
@@ -0,0 +1,269 @@
+                if (systemElement != null && systemElement.isJsonObject()) {
+                    JsonElement idElement = systemElement.getAsJsonObject().get("id");
+                    if (idElement != null && idElement.isJsonPrimitive()) {
+                        systemIdValue = idElement.getAsString();
+                    }
+                }
+            }
+        }
+        return Optional.ofNullable(systemIdValue);
+    }
+
+    /**
+     * Fetch the Live Data from the GridBox API by executing an API query.
+     *
+     * @param config The {@link GridBoxConfiguration} to use, containing a valid ID token
+     * @param responseHandler A function handling the retrieved live data, called if a valid response was received
+     *            with a {@link LiveData} object containing the content of the response.
+     * @throws IOException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws InterruptedException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws GridBoxApiException Thrown in case of an invalid response of the live data query, either
+     *             if a status code not equal to 200 or 403 is returned or the {@link LiveData} instance cannot be
+     *             parsed from the response body
+     * @throws GridBoxApiSystemNotFoundException Thrown in case the query returned status code 404, representing a
+     *             non-existing System ID
+     * @throws GridBoxApiAutheticationException Thrown in case of an invalid response of the authorization query if a
+     *             status code equal to 403 is returned
+     */
+    public void retrieveLiveData(GridBoxConfiguration config, Consumer<LiveData> responseHandler)
+            throws IOException, InterruptedException, GridBoxApiAutheticationException,
+            GridBoxApiSystemNotFoundException, GridBoxApiException {
+        // @formatter:off
+        HttpRequest liveDataRequest = HttpRequest
+                .newBuilder(URI.create("https://api.gridx.de/systems/%s/live".formatted(config.systemId)))
+                .GET()
+                .timeout(TIMEOUT_DURATION).setHeader("Content-Type", "application/json")
+                .setHeader("Authorization", "Bearer %s".formatted(config.idToken))
+                .build();
+        // @formatter:on
+
+        HttpResponse<String> response = HTTP_CLIENT.send(liveDataRequest, HttpResponse.BodyHandlers.ofString());
+        int status = response.statusCode();
+        String body = response.body();
+        if (status == 403) {
+            logger.warn("Access forbidden: status: {}, \n body: {}", status, body);
+            throw new GridBoxApiAutheticationException("Live Data request forbidden");
+        } else if (status == 404) {
+            logger.warn("Invalid response of live data request: status: {}, \n body: {}", status, body);
```

Comment:
Please adjust logging according to: https://www.openhab.org/docs/developer/guidelines.html#f-logging
change request metadata to debug and/or body contents to trace.

As the exception is thrown and bubbles up the chain to an thing status update, the event is allready logged by the framework, so warn should not be used here. Only debug/trace for request/response contents.